editor/state.js

```diff
@@ -96,7 +96,23 @@ export const blocks = combineUndoableReducers( {
 		}
 
 		return state;
-	}
+	},
+	dirty( state = false, action ) {
+		switch ( action.type ) {
+			case 'REPLACE_BLOCKS':
```

Comment:
After a rebase, this should be changed to "RESET_BLOCKS" and "REPLACE_BLOCKS" should be moved down.